doc/lsst.pipe.base/creating-a-task.rst

```diff
@@ -0,0 +1,350 @@
+
+Tasks are constructed by calling ``__init__`` with the :ref:`task configuration <creating-a-task-configuration>`.
+Occasionally additional arguments are required (see the task's documentation for details).
+`lsst.pipe.base.Task` has a few other arguments that are usually only specified when a task is created as a subtask of another task; you will probably never have to specify them yourself.
+(Command-line tasks are often constructed and run by calling `lsst.pipe.base.parseAndRun`, as described in :doc:`creating-a-command-line-task`.)
+
+Tasks typically have a ``run`` method that executes the task's main function.
+See :ref:`creating-a-task-class-methods` for more information.
+
+.. _creating-a-task-configuration:
+
+Configuration
+=============
+
+Every task requires a configuration: a task-specific set of configuration parameters.
+The configuration is read-only; once you construct a task, the same configuration will be used to process all data with that task.
+This makes the data processing more predictable: it does not depend on the order in which items of data are processed.
+
+The task's configuration is specified using a task-specific subclass of `lsst.pex.config.Config`.
+The task class specifies its configuration class using class variable `ConfigClass`.
+If the task has no configuration parameters then it may use `lsst.pex.config.Config` as its configuration class.
+
+Some important details of configurations:
+
+- Supply useful defaults for all config parameters if at all possible.
+  Your task will be much easier to use if the default configuration usually suffices.
+
+- Document each field of the configuration carefully.
+  Pretend you don't know anything about your task and ask yourself what you would need to be told to change the parameter.
+  What does the parameter do and why might you change it?
+  What units does it have?
+
+- Subtasks are specified in your configuration as fields of type `lsst.pex.config.ConfigurableField` or (less commonly) `lsst.pex.config.RegistryField`.
+  This allows subtasks to be :doc:`retargeted <command-line-task-retargeting-howto>` (replaced with a variant subtask).
+  For more information see :ref:`creating-a-task-subtasks`.
+
+  `~lsst.pipe.tasks.exampleStatsTasks.ExampleSigmaClippedStatsTask` uses configuration class `~lsst.pipe.tasks.exampleStatsTasks.ExampleSigmaClippedStatsConfig`:
+
+  .. code-block:: python
+
+     class ExampleSigmaClippedStatsConfig(pexConfig.Config):
+         """Configuration for ExampleSigmaClippedStatsTask.
+         """
+         badMaskPlanes = pexConfig.ListField(
+             dtype=str,
+             doc="Mask planes that, if set, indicate the associated pixel should "
+             "not be included when the calculating statistics.",
+             default=("EDGE",),
+         )
+         numSigmaClip = pexConfig.Field(
+             doc="number of sigmas at which to clip data",
+             dtype=float,
+             default=3.0,
+         )
```

Comment:
I'm surprised that this doesn't enforce `numSigmaClip > 0` as part of its definition